src/app/shared/scroll/scroll.directive.ts

```diff
@@ -0,0 +1,49 @@
+import {Directive, output} from '@angular/core';
+
+@Directive({
+    selector: '[appScroll]',
+    standalone: true,
+    host: {
+        '(scroll)': 'onScroll($event)',
+    },
+})
+export class ScrollDirective {
+    // eslint-disable-next-line @typescript-eslint/no-inferrable-types
+    private scrollTopSaved: number = 0;
+    readonly scrollLoadNextData = output<Event>();
+
+    onScroll(event: Event) {
+        // eslint-disable-next-line prefer-const
+        let scrollTopNew = (event.target as HTMLElement)?.scrollTop;
+        const scrollHeight = (event.target as HTMLElement)?.scrollHeight;
+        const clientHeight = (event.target as HTMLElement)?.clientHeight;
+
+        // eslint-disable-next-line @stylistic/padding-line-between-statements, @typescript-eslint/no-unused-vars
+        const scrollDownDirection: boolean | null =
+            scrollTopNew - this.scrollTopSaved === 0
+                ? null
+                : scrollTopNew - this.scrollTopSaved > 0;
```

Comment:
А не могли ли мы упростить условие до такого:
```suggestion
        const scrollDownDirection = scrollTopNew > this.scrollTopSaved;
```